### ISSUE TITLE
Errorr Connecting

### ISSUE THREAD

**Issue by MaxDevv** [mobile-shell/mosh#1136]
I run This command and get this error
sudo mosh username@server.ip --ssh="ssh -v -p 22"

OpenSSH_8.4p1 Ubuntu-5ubuntu1, OpenSSL 1.1.1j  16 Feb 2021
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Executing proxy command: exec '/usr/bin/mosh' '--family=prefer-inet' --fake-proxy -- server.ip 22
debug1: identity file /home/username/.ssh/id_rsa type 0
debug1: identity file /home/username/.ssh/id_rsa-cert type -1
debug1: identity file /home/username/.ssh/id_dsa type -1
debug1: identity file /home/username/.ssh/id_dsa-cert type -1
debug1: identity file /home/username/.ssh/id_ecdsa type -1
debug1: identity file /home/username/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/username/.ssh/id_ecdsa_sk type -1
debug1: identity file /home/username/.ssh/id_ecdsa_sk-cert type -1
debug1: identity file /home/username/.ssh/id_ed25519 type -1
debug1: identity file /home/username/.ssh/id_ed25519-cert type -1
debug1: identity file /home/username/.ssh/id_ed25519_sk type -1
debug1: identity file /home/username/.ssh/id_ed25519_sk-cert type -1
debug1: identity file /home/username/.ssh/id_xmss type -1
debug1: identity file /home/username/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.4p1 Ubuntu-5ubuntu1
debug1: Remote protocol version 2.0, remote software version OpenSSH_8.2p1 Ubuntu-4ubuntu0.2
debug1: match: OpenSSH_8.2p1 Ubuntu-4ubuntu0.2 pat OpenSSH* compat 0x04000000
debug1: Authenticating to server.ip:22 as 'username'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:ByUFhyNu2RR/fTe9Q5NktypouXfrgjfZr2o0FxLher0
debug1: Host 'server.ip' is known and matches the ECDSA host key.
debug1: Found key in /home/username/.ssh/known_hosts:7
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 134217728 blocks
debug1: Will attempt key: /home/username/.ssh/id_rsa RSA SHA256:f4d2ZXH9jMr9mh7TN6V/y++rPDs9ZFR agent
debug1: Will attempt key: /home/username/.ssh/id_dsa 
debug1: Will attempt key: /home/username/.ssh/id_ecdsa 
debug1: Will attempt key: /home/username/.ssh/id_ecdsa_sk 
debug1: Will attempt key: /home/username/.ssh/id_ed25519 
debug1: Will attempt key: /home/username/.ssh/id_ed25519_sk 
debug1: Will attempt key: /home/username/.ssh/id_xmss 
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,sk-ssh-ed25519@openssh.com,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ecdsa-sha2-nistp256@openssh.com>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Offering public key: /home/username/.ssh/id_rsa RSA SHA256:f4d2ZXH9jMr9mh7TN6V/y++rPDs9ZFR agent
debug1: Server accepts key: /home/username/.ssh/id_rsa RSA SHA256:f4d2ZXH9jMr9mh7TN6V/y++rPDs9ZFR agent
debug1: Authentication succeeded (publickey).
Authenticated to server.ip (via proxy).
debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: proc
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: Remote: /home/username/.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Remote: /home/username/.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
debug1: Sending environment.
debug1: Sending env LC_ADDRESS = en_US.UTF-8
debug1: Sending env LC_NAME = en_US.UTF-8
debug1: Sending env LC_MONETARY = en_US.UTF-8
debug1: Sending env LC_PAPER = en_US.UTF-8
debug1: Sending env LANG = en_NG.UTF-8
debug1: Sending env LC_IDENTIFICATION = en_US.UTF-8
debug1: Sending env LC_TELEPHONE = en_US.UTF-8
debug1: Sending env LC_MEASUREMENT = en_US.UTF-8
debug1: Sending env LC_TIME = en_US.UTF-8
debug1: Sending env LC_NUMERIC = en_US.UTF-8
debug1: Sending command: mosh-server 'new' '-c' '256' '-s' '-l' 'LANG=en_NG.UTF-8' '-l' 'LANGUAGE=en_NG' '-l' 'LC_NUMERIC=en_US.UTF-8' '-l' 'LC_TIME=en_US.UTF-8' '-l' 'LC_MONETARY=en_US.UTF-8' '-l' 'LC_PAPER=en_US.UTF-8' '-l' 'LC_NAME=en_US.UTF-8' '-l' 'LC_ADDRESS=en_US.UTF-8' '-l' 'LC_TELEPHONE=en_US.UTF-8' '-l' 'LC_MEASUREMENT=en_US.UTF-8' '-l' 'LC_IDENTIFICATION=en_US.UTF-8'
The locale requested by LANG=C.UTF-8 isn't available here.
Running `locale-gen C.UTF-8' may be necessary.

The locale requested by LANG=en_NG.UTF-8 isn't available here.
Running `locale-gen en_NG.UTF-8' may be necessary.

mosh-server needs a UTF-8 native locale to run.

Unfortunately, the local environment (LANG=C.UTF-8) specifies
the character set "US-ASCII",

The client-supplied environment (LANG=en_NG.UTF-8) specifies
the character set "US-ASCII".

debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0
locale: Cannot set LC_ALL to default locale: No such file or directory
LANG=en_NG.UTF-8
LANGUAGE=en_NG
LC_CTYPE="en_NG.UTF-8"
LC_NUMERIC=en_US.UTF-8
LC_TIME=en_US.UTF-8
LC_COLLATE="en_NG.UTF-8"
LC_MONETARY=en_US.UTF-8
LC_MESSAGES="en_NG.UTF-8"
LC_PAPER=en_US.UTF-8
LC_NAME=en_US.UTF-8
LC_ADDRESS=en_US.UTF-8
LC_TELEPHONE=en_US.UTF-8
LC_MEASUREMENT=en_US.UTF-8
LC_IDENTIFICATION=en_US.UTF-8
LC_ALL=
debug1: channel 0: free: client-session, nchannels 1
debug1: fd 2 clearing O_NONBLOCK
Connection to server.ip closed.
Transferred: sent 4436, received 4084 bytes, in 1.0 seconds
Bytes per second: sent 4324.8, received 3981.6
debug1: Exit status 1
/usr/bin/mosh: Did not find mosh server startup message. (Have you installed mosh on your server?)

**Comment by keithw** [comment]
We see this sometimes and it is generally a configuration problem unrelated to Mosh -- the server does not have a locale that is summoned by your locale-related environment variables (either the server's variables, or the client's variables). You will have the same problem when using SSH to remotely execute a command, although SSH allows the error to persist and for programs to produce buggy output rather than Mosh's strategy of bombing out with an error message.

Please see #1112, #793, #98.

Bottom line, to make this work you will probably have to set your client-side "LANG" environment variable to something supported on the server (which might be en_US.UTF-8 instead of en_NG.UTF-8), *or* build a Nigerian English locale on the server (possibly with the "locale-gen" command that is suggested, or the equivalent for your server OS).
